pkg/maintenance/alertmanagerMaintenance.go

```diff
@@ -94,13 +94,21 @@ func getAuthentication(c client.Client) (runtime.ClientAuthInfoWriter, error) {
 		return nil, err
 	}
 
-	var token string
+	token := ""
 	for _, s := range sl.Items {
 		if strings.Contains(s.Name, alertManagerServiceAccountName+"-token") {
 			token = string(s.Data["token"])
 		}
 	}
 
+	if len(token) == 0 {
+		// If serviceaccount token doesn't exist, try to use token request to get a token
+		token, err = metrics.RequestPrometheusServiceAccountAPIToken(c)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA due to %v", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to request token for prometheus-k8s SA due to %v", err)
```

Reply:
I used "get" here because here also means the read legacy-token failed to.